src/client/groups/GroupService.ts

```diff
@@ -0,0 +1,182 @@
+import { HDict, HGrid, HMarker, HRef, HaysonDict, HList } from 'haystack-core'
+import { ClientServiceConfig } from '../ClientServiceConfig'
+import { fetchVal } from '../fetchVal'
+import { dictsToGrid } from '../../util/hval'
+import { encodeQuery } from '../../util/http'
+import { ReadOptions } from '../RecordService'
+
+/**
+ * a group record
+ */
+export interface Group extends HDict {
+	id?: HRef
+	userGroup: HMarker
+}
+
+/**
+ * optional parameters available to readByFilter query
+ */
+export type GroupsReadOptions = Omit<ReadOptions, 'unique'>
+
+export class GroupsService<T extends Group = Group> {
```

Comment:
Should have class level comment describing what this is for.

Reply:
Added.